### PR TITLE
`azurerm_network_manager` - support management group register provider

### PR DESCRIPTION
**Pull request by teowa**
- Register `Microsoft.Network` provider at Management Group scope if the network manager use MG as scope.
- Parallelize the tests of network manager using Management Group as scope. Before this one subscription can provision only one network manager at a time.
- Sleep 10 seconds to make sure provider is successfully registered, related to https://github.com/Azure/azure-rest-api-specs/issues/22041.

### CODE REVIEW

File: internal/services/network/network_manager_resource.go [hunk 1]
```diff
@@ -175,6 +176,23 @@ func (r ManagerResource) Create() sdk.ResourceFunc {
...
+				for _, group := range state.Scope[0].ManagementGroups {
+					groupId, err := managementGroupParse.ManagementGroupID(group)
+					if err != nil {
+						return err
+					}
+					_, err = resourceProvidersClient.RegisterAtManagementGroupScope(ctx, "Microsoft.Network", groupId.Name)
+					if err != nil {
+						return fmt.Errorf("registering the 'Microsoft.Network' provider with Management Group %s: %+v", groupId, err)
+					}
+				}
```
Comment (tombuildsstuff) [inline internal/services/network/network_manager_resource.go:191]: can you give some more context on the issue here? Terraform shouldn't be silently registering resource providers at management group scopes which generally require extra-permissions

File: internal/services/network/network_manager_resource.go [hunk 2]
```diff
@@ -175,6 +176,23 @@ func (r ManagerResource) Create() sdk.ResourceFunc {
...
+				// https://github.com/Azure/azure-rest-api-specs/issues/22041
+				time.Sleep(10 * time.Second)
```
Comment (tombuildsstuff) [inline internal/services/network/network_manager_resource.go:193]: Resource Provider registration takes up to 15 minutes, and unfortunately `time.Sleep`'s aren't something we can ship/rely on - as such I think we can remove all of this RP auto-registration logic here?

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hey @teowa 

Thanks for this PR.

Unfortunately this PR contains a number of design issues which are going to need to be addressed before we can proceed here - namely registering Resource Providers at the Management Group level generally requires elevated permissions which may not be available, in addition the approach used here is problematic (for example, what happens if the RP isn't registered in 10s [since this can take up to 15 minutes] - what happens if we attempt to register multiple in parallel etc).

Once those issues are addressed we can take another look here, but I think it'd be helpful to have some additional context on what the problem is that we're looking to solve here.

Thanks!

**Comment by teowa** [comment]
Provisioning network manager using MG as scope need the 'Microsoft.Network' provider being registered at MG level, and using MG to provision network manager can parallelize the acc tests, see https://github.com/hashicorp/terraform-provider-azurerm/pull/19593#issuecomment-1352497602. But seems we have to close this PR due to lack of related method, https://github.com/Azure/azure-rest-api-specs/issues/22041.

**Pull request closed without merging**